fix(orders): refine order update process and make fields optional

sebasx200 opened the pull request:
- Make order update fields (summary, status, due date) optional instead of required
- Remove ability to change order vehicle after creation
- Add dynamic calculation of due date based on specified days
- Simplify repository update method to rely on EF Core change tracking
- Ensure proper null-checking for optional update fields
- Changed http method to patch in the controller for updating

jpmvalencia commented:
Approved. I've reviewed the changes and everything is very neat. This PR includes the vehicle list when the vehicle is created.

Pretty good job.

sebasx200 committed:
fix(orders): refine order update process and make fields optional

- Make order update fields (summary, status, due date) optional instead of required
- Remove ability to change order vehicle after creation
- Add dynamic calculation of due date based on specified days
- Simplify repository update method to rely on EF Core change tracking
- Ensure proper null-checking for optional update fields
- Changed http method to patch in the controller for updating

sebasx200 commented:
merge accepted
